### ISSUE TITLE
Imvertor implementeert paragraaf '5.16.1 Uniekheid van namen van modelelementen' van MIM 1.1.1 niet volledig.

### ISSUE THREAD

**Issue by melsk-r** [Imvertor/Imvertor-Maven#493]
Volgens paragraaf '5.16.1 Uniekheid van namen van modelelementen' mogen er geen 2 Objecttypen met dezelfde naam binnen een Informatiemodel bestaan. Hetzelfde geldt voor een Datatype. Wel mag de voor een Objecttype gebruikte naam nogmaals gebruikt worden voor den Datatype.

Deze regel is niet goed geïmplementeerd in Imvertor want indien deze situatie zich voordoet wordt er geen waarschuwing gedefinieerd.
Overigens ben ik van mening dat dit tot een error zou moeten leiden.

Zie voor een voorbeeld de job '2024-06-06-14-36-37-920' m.b.t. 'SIM/Inkomen Modellen/20230101'. Daarin bestaan 2 Objecttypes met de naam 'Vrijlating inkomsten'.

**Comment by ArjanLoeffen** [comment]
De job [2024-06-06-14-36-37-920](https://imvertor.armatiek.nl/imvertor-executor/dashboard/app?job=2024-06-06-14-36-37-920) is niet meer toegankelijk. 

Als _Vrijlating inkomsten_ tweemaal voorkomt in één domein is dat m.i. fout. Wordt daar niet op gesignalleerd?

**Comment by melsk-r** [comment]
Nee, in de vermeldde job was dat i.i.g. niet het geval.

**Comment by ArjanLoeffen** [comment]
Eigenlijk ben ik van mening dat het nog strakker moet: we zouden dit soort situaties moeten tegenhouden. Namen behoren uniek te zijn, in ieder geval binnen hetzelfde domein. Dus dat binnen één domein _Vrijlating inkomsten_ als objecttype én enumeratie oid voorkomt, zou moeten worden gesignaleerd als fout. Maar ja, de MIM standaard zegt er wrsch niks over.

**Comment by melsk-r** [comment]
> Eigenlijk ben ik van mening dat het nog strakker moet: we zouden dit soort situaties moeten tegenhouden. Namen behoren uniek te zijn, in ieder geval binnen hetzelfde domein. Dus dat binnen één domein _Vrijlating inkomsten_ als objecttype én enumeratie oid voorkomt, zou moeten worden gesignaleerd als fout. Maar ja, de MIM standaard zegt er wrsch niks over.

Het lijkt er op dat wij het dan eens zijn.

> Overigens ben ik van mening dat dit tot een error zou moeten leiden.

**Comment by ArjanLoeffen** [comment]
Ja, was een bug. Heb het hersteld: Namen behoren uniek te zijn, in ieder geval binnen hetzelfde domein.
